jsweet_extension/io/apicurio/PromiseAdapter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apicurio;
+
+import java.util.List;
+import org.jsweet.transpiler.extension.PrinterAdapter;
+import org.jsweet.transpiler.model.MethodInvocationElement;
+import org.jsweet.transpiler.model.NewClassElement;
+
+import java.util.concurrent.CompletableFuture;
+
+import javax.lang.model.element.Element;
+
+/**
+ * This adapter provides transpilation from {@link java.util.concurrent.CompletableFuture}
+ * and its methods to <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise">JavaScript Promise</a>
+ */
+public class PromiseAdapter extends PrinterAdapter {
```

Comment:
As far as I can tell this needs to be in the project root along with `jsweetconfig.json`. I did try and put them in `src/main/java` but it failed to transpile.

A side-effect of it being in the root is there isn no Intellisense for this file.